glean-core/src/lib.rs

```diff
@@ -446,10 +446,15 @@ impl Glean {
     /// A deletion_request ping is sent, all pending metrics, events and queued
     /// pings are cleared, and the client_id is set to KNOWN_CLIENT_ID.
     /// Afterward, the upload_enabled flag is set to false.
-    fn on_upload_disabled(&mut self) {
+    fn on_upload_disabled(&mut self, during_init: bool) {
         // The upload_enabled flag should be true here, or the deletion ping
         // won't be submitted.
-        if let Err(err) = self.internal_pings.deletion_request.submit(self, None) {
+        let reason = if during_init {
+            Some("at_init")
+        } else {
+            Some("set_upload_enabled")
+        };
```

Comment:
Soon:

```suggestion
        let reason = during_init.then(|| "at_init").or(Some("set_upload_enabled"));
```

[do not merge -- this is Rust 1.51.0 and we support lower Rust versions for now]

Reply:
Looks kinda messy, honestly. The return type of the closure doesn't match the type of the `or`... maybe

```rust
let reason = during_init.then(|| "at_init").or_else(|| "set_upload_enabled");
```

Reply:
still needs `.or_else(|| Some("set_upload_enabled")` really ;)
(anyway, we can't use that anyway)